store/postgres_graphql.go

```diff
@@ -284,14 +314,100 @@ func psqlSubQuery(tenant string, graph *schemaGraph, qb *sqlQueryBuilder, tc *ta
 			return err
 		}
 		tc.children = append(tc.children, subColumns)
 	}
+
+	// TODO: make error messages more helpful (and more systematic) in the following code
+	// Process order_by, if available. At this point all table aliases are known.
+	if orderByArg != nil {
+
+		arg := orderByArg
+
+		// This argument's value is a slice of values representing (table, field, order) tuples.
+		orderByItems, ok := (arg.Value.GetValue()).([]ast.Value)
+		if !ok {
+			return fmt.Errorf("argument type is wrong in `order_by`: %s : %#v", arg.Name.Value, arg.Value.GetValue())
+		}
+
+		// Process each of the {table, field, order} objects...
+		for _, item := range orderByItems {
+
+			fields, ok := item.GetValue().([]*ast.ObjectField)
+			if !ok {
+				return fmt.Errorf("failed to type cast argument value for `order_by`, step 2, value: %#v", item.GetValue())
+			}
+
+			// FIXME probably best to make it a proper type {table,field, order}
+			ob := map[string]string{}
+
+			// Process a single {table, field, order} entry
+			for _, objectField := range fields {
+
+				switch objectField.Name.Value {
+				case "table":
+					s, ok := objectField.GetValue().(*ast.StringValue)
+					if !ok {
+						return fmt.Errorf("failed to type cast argument value for `order_by`, step 3: %#v: %#v", objectField.Name.Value, objectField)
+					}
+					ob["table"], _ = s.GetValue().(string) // FIXME: is this safe?
+
+				case "field":
+					s, ok := objectField.GetValue().(*ast.StringValue)
+					if !ok {
+						return fmt.Errorf("failed to type cast argument value for `order_by`, step 3, value: %#v", objectField)
+					}
+					ob["field"] = s.GetValue().(string) // FIXME: is this safe?
+
+				case "order":
+					s, ok := objectField.GetValue().(*ast.StringValue)
+					if !ok {
+						return fmt.Errorf("failed to type cast argument value for `order_by`, step 3, value: %#v", objectField)
+					}
+					ob["order"] = s.GetValue().(string) // FIXME: is this safe?
+				}
+			}
+
+			tableName := ob["table"]
+			f := ob["field"]
+			order := strings.ToUpper(ob["order"])
+
+			if order != orderAsc && order != orderDesc {
+				return fmt.Errorf("invalid sorting order for: %s", ob["order"])
+			}
+
+			// Lookup the table alias by table name
+			ta := findAliasFor(tableName, root)
+			if ta == "" {
+				return fmt.Errorf("could not find the table alias for: %s", tableName)
+			}
+
+			// Augment the SQL query with the ORDER BY statement.
+			qb.sql = qb.sql.OrderBy(tableColumn(ta, f) + " " + order)
+		}
+	}
+
 	return nil
 }
 
+// findAliasFor takes a table name and traverses the tableColumns tree recursively, looking for the first match
+// on the table name. On match, it returns the stored table name alias for that table name.
+// If the the table name cannot be found in the tree, the empty string is returned.
+func findAliasFor(tableName string, tc *tableColumns) string {
+	if tc.table == tableName {
+		return tc.alias
+	} else {
```

Comment:
Just for readability, this else is not required. The block could just be flat in the function

Reply:
I agree. I've added this to my TODO list.